### PR TITLE
Created vector() method as property in the Doc class

### PR DESCRIPTION
**Pull request by bicycleman15**
I just added tests to the already done work by @prashantksharma in his PR #21 
resloves issue #19

### CODE REVIEW

File: tests/local_mode/test_doc_vector.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import syft as sy
```
Comment (AlanAboudib) [inline tests/local_mode/test_doc_vector.py]: can you please add those function inside a file called `test_doc.py`

File: syfertext/doc.py [hunk 2]
```diff
@@ -112,21 +112,17 @@ def __iter__(self):
...
+        Get document vector as an average of in-vocabulary token's vectors 
+        
         Args:
-            self (Doc): current document.
-            workers (sequence of BaseWorker): A sequence of remote workers from .
-            crypto_provider (BaseWorker): A remote worker responsible for providing cryptography (SMPC encryption) functionalities.
-            requires_grad (bool): A boolean flag indicating whether gradients are required or not.
-
+          self (Doc): current document
+
```
Comment (AlanAboudib) [inline syfertext/doc.py]: no need to document what `self` is

File: syfertext/doc.py [hunk 3]
```diff
@@ -136,22 +132,45 @@ def get_encrypted_vector(self, *workers, crypto_provider=None, requires_grad=Tru
...
+        return doc_vector
+
+    def get_encrypted_vector(
+        self, *workers, crypto_provider=None, requires_grad=True,
+    ):
+        """Get the mean of the vectors of each Token in this documents.
+
+        Args:
+            self (Doc): current document.
+            workers (sequence of BaseWorker): A sequence of remote workers from .
```
Comment (AlanAboudib) [inline syfertext/doc.py]: no need to document `self`

File: syfertext/doc.py [hunk 4]
```diff
@@ -136,22 +132,45 @@ def get_encrypted_vector(self, *workers, crypto_provider=None, requires_grad=Tru
...
         else:
             # Create the final Doc vector
             doc_vector = vectors / vector_count
 
+        return doc_vector
+
+    def get_encrypted_vector(
+        self, *workers, crypto_provider=None, requires_grad=True,
+    ):
+        """Get the mean of the vectors of each Token in this documents.
```
Comment (AlanAboudib) [inline syfertext/doc.py]: Please add type hinting, we use the `typing` library for this:

example

```
crypto_provide: BaseWorker = None
requires_grad: bool = True,
```

Comment (bicycleman15) [reply syfertext/doc.py]: Should I also add type hinting for *workers as
`*workers : List[BaseWorker]`
??

**Comment by bicycleman15** [comment]
Please have a look @AlanAboudib

**Comment by prashantksharma** [comment]
@bicycleman15 Thank you so much!

File: tests/local_mode/test_doc.py [hunk 5]
```diff
@@ -2,12 +2,66 @@
...
+    """Test that the average vector of invalid tokens match the expected result"""
+    doc = nlp("non-valid-token")
+    actual = doc.vector
+    vectors = None
+
+    # Count the tokens that have vectors
+    vector_count = 0
+    for token in doc:
+        # Get the encypted vector of the token if one exists
+        if token.has_vector:
```
Comment (AlanAboudib) [inline tests/local_mode/test_doc.py]: you are not getting the encrypted vectors here, these are the plaintext vectors

File: tests/local_mode/test_doc.py [hunk 6]
```diff
@@ -2,12 +2,66 @@
...
+
 hook = sy.TorchHook(torch)
 me = hook.local_worker
 
 nlp = syfertext.load("en_core_web_lg", owner=me)
 
 
+def test_avg_vector_valid_token():
+    """Test that the average vector of valid tokens match the expected result"""
+    doc = nlp("on your left")
```
Comment (AlanAboudib) [inline tests/local_mode/test_doc.py]: Please make your code eye-friendly by inserting new lines between code blocks.

File: syfertext/doc.py (hunk 4 above)
Comment (AlanAboudib) [reply syfertext/doc.py]: @bicycleman15 it would be enought as `*workers: BaseWorker`

File: syfertext/doc.py [hunk 7]
```diff
@@ -121,21 +121,14 @@ def __iter__(self):
             # Yield a Token object
             yield self[i]
 
-    def get_encrypted_vector(self, *workers, crypto_provider=None, requires_grad=True):
-        """Get the mean of the vectors of each Token in this documents.
-
-        Args:
-            self (Doc): current document.
-            workers (sequence of BaseWorker): A sequence of remote workers from .
```
Comment (AlanAboudib) [inline syfertext/doc.py]: Please remove the document of `self` from the the `Args:` section

File: tests/local_mode/test_doc.py (hunk 5 above)
Comment (bicycleman15) [reply tests/local_mode/test_doc.py]: right , silly mistake

**Pull request merged**